src/public/services/AdminSubmissionsService.ts

```diff
@@ -65,3 +68,43 @@ export const getEncryptedResponse = ({
     .get(`${ADMIN_FORM_ENDPOINT}/${formId}/submissions/${submissionId}`)
     .then(({ data }) => data)
 }
+
+/**
+ * Triggers a download of a set of attachments as a zip file when given attachment metadata and a secret key
+ * @param {Map} attachmentDownloadUrls Map of question number to individual attachment metadata (object with url and filename properties)
+ * @param {String} secretKey An instance of EncryptionKey for decrypting the attachment
+ * @returns {Promise} A Promise containing the contents of the ZIP file as a blob
+ */
+export const downloadAndDecryptAttachmentsAsZip = async (
+  attachmentDownloadUrls: Map<number, { url: string; filename: string }>,
+  secretKey: string,
+): Promise<Blob> => {
+  const zip = new JSZip()
+  const downloadPromises = Array.from(attachmentDownloadUrls).map(
+    async ([questionNum, { url, filename }]) => {
+      const bytesArray = await downloadAndDecryptAttachment(url, secretKey)
```

Comment:
don't mix async-await syntax with Promise.all, Promise.then etc. this works but is unnecessarily confusing. this case is particularly confusing because you're mixing `map` with `async`, which can lead to [unexpected behaviour](https://stackoverflow.com/questions/40140149/use-async-await-with-array-map) (the same is the case for [forEach](https://stackoverflow.com/questions/37576685/using-async-await-with-a-foreach-loop)).

I'd much prefer it if we could stick to one or the other, whichever is more appropriate for the case. in this case we can probably stick to chaining the Promises.

Reply:
ps @mantariksh, gonna need you to explain abit more on this - my understanding of this is that we don't mix `async/await` and `then/catch` in promises because they are equivalent (as `async/await` is just syntactic sugar over `then/catch`). 

however, because `Promise.all` is a static method and doesn't have an equivalent, wouldn't we need to use `Promises` every time we need the static methods of the `Promise` object?

Reply:
>wouldn't we need to use Promises every time we need the static methods of the Promise object?

yeah exactly. when you use `Promise.all`, you have to generate an array of Promises, so normally you wouldn't have to `await` on anything in process anyway.

I suppose it's possible that this is just a me thing. like I said, the code works anyway. I'll leave this to your judgement, if you think it's fine then I'm happy to go with it!